frontend/src/app/modules/user/profile-page-main-info/profile-page-main-info.component.ts

```diff
@@ -47,4 +67,65 @@ export class ProfilePageMainInfoComponent implements OnChanges {
     private updateSolutions() {
         this.communityLastWeekSolution = utils.getLastWeekCount(this.user.solutions);
     }
+
+    private getCurrentUser() {
+        this.authService.getUser().subscribe((user: IUser) => {
+            this.currentUser = user;
+        });
+    }
+
+    addFriend() {
+        this.getCurrentUser();
+
+        const request: INewFriendship = {
+            senderId: this.currentUser?.id as number,
+            recipientId: this.user.id,
+        };
+
+        this.userService
+            .sendFriendshipRequest(request)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next: () => {
+                    this.toastrNotification.showSuccess('Friendship request sent successfully.');
+                },
+                error: (error) => {
+                    this.toastrNotification.showError(error);
+                },
+            });
+    }
+
+    deleteFriend() {
+        this.getCurrentUser();
```

Comment:
why do we need to reload the current user on every action? load user since in OnInit and use everythere